Formula/trojan-go.rb

```diff
@@ -0,0 +1,105 @@
+  sha256 "423fa05d81da0105a01682a77bbbd95fef09644c3a9dbbddd641436e31879429"
+  license "GPL-3.0-only"
+  head "https://github.com/p4gefau1t/trojan-go.git"
+
+  bottle do
+  end
+
+  depends_on "go" => :build
+
+  resource "geoip" do
+    url "https://github.com/v2fly/geoip/releases/download/202105070029/geoip.dat"
+    sha256 "f3aeb80c21fd403c6030988477d0cac7574322b786733d0ecbc24ff4ab55ce93"
+  end
+
+  resource "geosite" do
+    url "https://github.com/v2fly/domain-list-community/releases/download/20210507022421/dlc.dat"
+    sha256 "c08b57f45ad248e3812ff6c453c788c85916cd443d4c11e26ccd1bb6919c928d"
+  end
+
+  def install
+    execpath = libexec/name
+    system "go", "build", *std_go_args, "-o", execpath, "-tags=full"
+    (bin/"trojan-go").write_env_script execpath,
+      TROJAN_GO_LOCATION_ASSET: "${TROJAN_GO_LOCATION_ASSET:-#{pkgshare}}"
+
+    pkgetc.install "example/client.json" => "config.json"
+
+    resource("geoip").stage do
+      pkgshare.install "geoip.dat"
+    end
+
+    resource("geosite").stage do
+      pkgshare.install "dlc.dat" => "geosite.dat"
+    end
+  end
+
+  plist_options manual: "trojan-go -config #{HOMEBREW_PREFIX}/etc/trojan-go/config.json"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+        <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+        <plist version="1.0">
+        <dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{bin}/trojan-go</string>
+            <string>-config</string>
+            <string>#{etc}/trojan-go/config.json</string>
+          </array>
+          <key>KeepAlive</key>
+          <true/>
+          <key>RunAtLoad</key>
+          <true/>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    trojan_go_server_port = free_port
+    (testpath/"server.yaml").write <<~EOS
+      run-type: server
+      local-addr: 127.0.0.1
+      local-port: #{trojan_go_server_port}
+      remote-addr: 127.0.0.1
+      remote-port: 80
+      password:
+        - test
+      ssl:
+        cert: cert.crt
+        key: key.key
+        sni: www.domain-name.com
+    EOS
+    server = fork { exec "#{bin}/trojan-go", "-config", testpath/"server.yaml" }
+
+    trojan_go_client_port = free_port
+    (testpath/"client.yaml").write <<~EOS
+      run-type: client
+      local-addr: 127.0.0.1
+      local-port: #{trojan_go_client_port}
+      remote-addr: 127.0.0.1
+      remote-port: #{trojan_go_server_port}
+      password:
+        - test
+    EOS
+    client = fork { exec "#{bin}/trojan-go", "-config", testpath/"client.yaml" }
+
+    sleep 3
+    begin
+      system "curl", "--socks5", "127.0.0.1:#{trojan_go_client_port}", "github.com"
+    ensure
+      Process.kill 9, server
+      Process.wait server
+      Process.kill 9, client
+      Process.wait client
+    end
+  end
```

Comment:
I think the better way to test it is to run `trojan-go -version`, and match the output.

Reply:
Not a maintainer, but submitting only a version test is highly likely to get your formula rejected: https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
`test` is used to test the function of the software, but it cannot test whether the software is working normally.
If there is any error in the build, it will not be able to test whether the software is working properly.
For example, if you remove the `-tags=full`:
`$ trojan-go -version` shows the same output as normal one.